Source/automap.cpp

```diff
@@ -96,161 +96,178 @@ void DrawDiamond(const Surface &out, Point center, uint8_t color)
 		return;
 	}
 	if (tile.HasFlag(AutomapTile::Flags::VerticalDoor)) { // two wall segments with a door in the middle
-		DrawMapVerticalDoor(out, { center.x - AmLine16, center.y - AmLine8 });
+		DrawMapVerticalDoor(out, { center.x - AmLine16, center.y - AmLine8 }, colorBright, colorDim);
 	}
 	if (tile.HasFlag(AutomapTile::Flags::VerticalGrate)) { // right-facing half-wall
-		DrawMapLineNE(out, { center.x - AmLine32, center.y }, AmLine8, MapColorsDim);
-		DrawDiamond(out, { center.x, center.y - AmLine8 }, MapColorsDim);
+		DrawMapLineNE(out, { center.x - AmLine32, center.y }, AmLine8, colorDim);
+		DrawDiamond(out, { center.x, center.y - AmLine8 }, colorDim);
 	} else if (tile.HasFlag(AutomapTile::Flags::VerticalArch)) { // window or passable column
-		DrawDiamond(out, { center.x, center.y - AmLine8 }, MapColorsDim);
+		DrawDiamond(out, { center.x, center.y - AmLine8 }, colorDim);
 	}
 }
 
 /**
  * For caves the horizontal/vertical flags are swapped
  */
-void DrawCaveHorizontal(const Surface &out, Point center, AutomapTile tile)
+void DrawCaveHorizontal(const Surface &out, Point center, AutomapTile tile, uint8_t colorBright, uint8_t colorDim)
 {
 	if (tile.HasFlag(AutomapTile::Flags::VerticalDoor)) {
-		DrawMapHorizontalDoor(out, { center.x - AmLine16, center.y + AmLine8 });
+		DrawMapHorizontalDoor(out, { center.x - AmLine16, center.y + AmLine8 }, colorBright, colorDim);
 	} else {
-		DrawMapLineSE(out, { center.x - AmLine32, center.y }, AmLine16, MapColorsDim);
+		DrawMapLineSE(out, { center.x - AmLine32, center.y }, AmLine16, colorDim);
 	}
 }
 
 /**
  * For caves the horizontal/vertical flags are swapped
  */
-void DrawCaveVertical(const Surface &out, Point center, AutomapTile tile)
+void DrawCaveVertical(const Surface &out, Point center, AutomapTile tile, uint8_t colorBright, uint8_t colorDim)
 {
 	if (tile.HasFlag(AutomapTile::Flags::HorizontalDoor)) {
-		DrawMapVerticalDoor(out, { center.x + AmLine16, center.y + AmLine8 });
+		DrawMapVerticalDoor(out, { center.x + AmLine16, center.y + AmLine8 }, colorBright, colorDim);
 	} else {
-		DrawMapLineNE(out, { center.x, center.y + AmLine16 }, AmLine16, MapColorsDim);
+		DrawMapLineNE(out, { center.x, center.y + AmLine16 }, AmLine16, colorDim);
 	}
 }
 
+AutomapTile GetAutomapTypeView(Point map);
+
 /**
  * @brief Renders the given automap shape at the specified screen coordinates.
  */
-void DrawAutomapTile(const Surface &out, Point center, AutomapTile tile)
+void DrawAutomapTile(const Surface &out, Point center, Point map)
 {
+	AutomapTile tile = GetAutomapTypeView(map);
+	uint8_t colorBright = MapColorsBright;
+	uint8_t colorDim = MapColorsDim;
+	MapExplorationType explorationType = static_cast<MapExplorationType>(AutomapView[map.x][map.y]);
+
+	switch (explorationType) {
+	case MapExplorationType::MAP_EXP_SHRINE:
+		colorDim = PAL16_BLUE + 8;
+		colorBright = PAL8_BLUE;
+		break;
+	case MapExplorationType::MAP_EXP_SELF:
+	case MapExplorationType::MAP_EXP_NONE:
+		break;
+	}

```

Comment:
I'd recommend extracting this whole switch into a function and returning the colors from there.

Right now we have 2 different places setting the colors in here and in lines 216/217. We should centralize that in the switch.

Reply:
nah, it's good, this is the only function that needs the colors, no need to artifically create a new one then mess around with returning 2 values and other stuff